vignettes/archetypes.Rmd

```diff
@@ -0,0 +1,343 @@
+
+
+```r
+attr(archetype, "brm_archetype_nuisance")
+#> [1] "nuisance_biomarker1"          "nuisance_biomarker2"          "nuisance_status1_absent"     
+#> [4] "nuisance_status2_present"     "nuisance_baseline"            "nuisance_baseline.timetime_2"
+#> [7] "nuisance_baseline.timetime_3"
+```
+
+The factors of interest linearly map to marginal means. To see the mapping, call `summary()` on the archetype. The printed output helps build intuition on how the archetype is parameterized and what those parameters are doing.
+
+
+```r
+summary(archetype)
+#> # This object is an informative prior archetype in brms.mmrm.
+#> # The fixed effect parameters of interest express the
+#> # marginal means as follows (on the link scale):
+#> # 
+#> #    group_1:time_2 = x_group_1_time_2
+#> #    group_1:time_3 = x_group_1_time_2 + x_group_1_time_3
+#> #    group_1:time_4 = x_group_1_time_2 + x_group_1_time_3 + x_group_1_time_4
+#> #    group_2:time_2 = x_group_2_time_2
+#> #    group_2:time_3 = x_group_2_time_2 + x_group_2_time_3
+#> #    group_2:time_4 = x_group_2_time_2 + x_group_2_time_3 + x_group_2_time_4
+```
+
+## Informative priors
+
+In the printed output from `summary(archetype)`, parameters of interest such as `x_group_1_time_2` and `x_group_2_time_3` are always labeled using treatment groups and time points in the data (and subgroup levels, if applicable). Even though different archetypes have different parameterizations and thus different ways of expressing marginal means, this labeling scheme remains consistent across all archetypes. `brms.mmrm` uses this to help you assign priors. First, match your priors to levels in the data.
+
+
+```r
+label <- brm_prior_label(
+  code = "normal(1, 2.2)",
+  group = "group_1",
+  time = "time_2"
+) |>
+  brm_prior_label("normal(1, 3.3)", group = "group_1", time = "time_3") |>
+  brm_prior_label("normal(1, 4.4)", group = "group_1", time = "time_4") |>
+  brm_prior_label("normal(2, 2.2)", group = "group_2", time = "time_2") |>
+  brm_prior_label("normal(2, 3.3)", group = "group_2", time = "time_3") |>
+  brm_prior_label("normal(2, 4.4)", group = "group_2", time = "time_4")
+label
+#> # A tibble: 6 × 3
+#>   code           group   time  
+#>   <chr>          <chr>   <chr> 
+#> 1 normal(1, 2.2) group_1 time_2
+#> 2 normal(1, 3.3) group_1 time_3
+#> 3 normal(1, 4.4) group_1 time_4
+#> 4 normal(2, 2.2) group_2 time_2
+#> 5 normal(2, 3.3) group_2 time_3
+#> 6 normal(2, 4.4) group_2 time_4
+```
+
+Those `group` and `time` labels map your priors to the corresponding `x_*` parameters.
+
+
+```r
+prior <- brm_prior_archetype(label = label, archetype = archetype)
+prior
```

Comment:
I here wondered about what some of the columns (`group`, `resp`, `dpar`,...) are there for. It seems they are needed for compatibility with `{brms}` (`set_prior()`). This is exlpained in the help files; perhaps also clarify it here? Please ignore if I have overlooked something.

Reply:
I added a bit of explanation in 9daec5528e33b03d3007b2d1575febd78b9f8602. `brm_prior_archetype()` internally calls `brms::set_prior()` and returns a `brms` prior object exactly as documented in https://paul-buerkner.github.io/brms/reference/set_prior.html. I am not sure what `group` and `resp` really do, but `dear` is for distributional models as explained at https://cran.r-project.org/web/packages/brms/vignettes/brms_distreg.html, which we use for `sigma`.